pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -255,7 +267,7 @@ func resolveSymlink(fileinfo os.FileInfo, path string) (os.FileInfo, error) {
 	return fileinfo, err
 }
 
-func createWalkFn(filesOnDisk map[string]os.FileInfo) filepath.WalkFunc {
+func createWalkFn(filesOnDisk map[string]os.FileInfo, fileFilterFunc func(fileInfo os.FileInfo) bool) filepath.WalkFunc {
```

Comment:
One alternative to adding `fileFilterFunc` argument would be to define `createWalkFn` to variable, like `var createWalkFn =  createWalkFn(filesOnDisk map[string]os.FileInfo) filepath.WalkFunc` allowing that to be overridden in tests only. Maybe `fileFilterFunc` is used outside tests? What do you think?

Reply:
I was thinking about just mocking the `createWalkFn` but I wanted to minimize the code duplication, and there is some other stuff in `createWalkFn` that I would have to duplicate in the mock to test properly. Also if possible I tend to prefer injecting dependencies rather than overwriting global factory functions which than need to reset and can have other implications that are less visible (like global factory function used in multiple contexts but mocked only for one of them).

At the moment `fileFilterFunc` is used only in tests and is private (on struct and also as `createWalkFn` is private). I mean there could be usecase for it outside in which case it could be made an argument in the `NewDashboardFileReader` factory function.

Reply:
Understood. My thought was that the createWalkFn functionality was tested in other tests and you only are using the fileFilterFunc in one other test so just changing createWalkFn implementation would have been easier to reason about since it wouldn't require you to send in `nil`as `fileFilterFunc` in other tests. But this is fine I suppose.

Reply:
In the end I refactored the test so I do not need that and so I removed it completely.